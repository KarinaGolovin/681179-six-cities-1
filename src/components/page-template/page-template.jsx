import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Header} from '../header/header.jsx';
import {getAuthorizationStatus} from '../../store/actions';
import {Footer} from '../footer/footer.jsx';

export class PageTemplate extends PureComponent {
  render() {
    const {children, isAuthorizationRequired, user} = this.props;
    return (
      <>
        <Header
          isAuthorizationRequired={isAuthorizationRequired}
          user={user}
        />
        {children}
        <Footer />
      </>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    isAuthorizationRequired: getAuthorizationStatus(state),
    user: {
      id: state.user.id,
      email: state.user.email,
      name: state.user.name,
      avatar: state.user.avatar_url,
      isPro: state.user.is_pro
    }
  };
};

PageTemplate.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number,
    email: PropTypes.string,
    name: PropTypes.string,
    avatar: PropTypes.string,
    isPro: PropTypes.bool
  }),
  isAuthorizationRequired: PropTypes.bool,
  children: PropTypes.any
};

export default connect(
    mapStateToProps
)(PageTemplate);

