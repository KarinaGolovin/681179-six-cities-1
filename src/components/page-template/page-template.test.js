import React from 'react';
import {PageTemplate} from './page-template.jsx';
import {Router} from 'react-router-dom';
import history from '../../history';
import ShallowRenderer from 'react-test-renderer/shallow';

const renderer = new ShallowRenderer();

const mockUser = {
  id: 1,
  email: `test@test.com`,
  name: `Test`,
  avatar: ``,
  isPro: false,
};

it(`PageTemplate correctly renders`, () => {
  const component = renderer.render(
      <Router history={history}>
        <PageTemplate
          isAuthorizationRequired={false}
          user={mockUser}
        />
      </Router>
  );

  expect(component).toMatchSnapshot();
});
