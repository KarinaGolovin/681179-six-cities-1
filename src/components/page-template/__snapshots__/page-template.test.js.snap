// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTemplate correctly renders 1`] = `
<Context.Provider
  value={
    Object {
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "match": Object {
        "isExact": true,
        "params": Object {},
        "path": "/",
        "url": "/",
      },
      "staticContext": undefined,
    }
  }
>
  <PageTemplate
    isAuthorizationRequired={false}
    user={
      Object {
        "avatar": "",
        "email": "test@test.com",
        "id": 1,
        "isPro": false,
        "name": "Test",
      }
    }
  />
</Context.Provider>
`;
